jupyter_accessibility/dodo.py

```diff
@@ -46,87 +33,75 @@ def remove_directory(*dir):
-
-    def main(self, args=None, standalone=False):
-        """invoke the doit application through an interactive api."""
-        try:
-            self.project.doit().run(prep_args(args))
-        except standalone and SystemExit or () as e:
-            if e.args[0] != 0:
-                raise e
-
-    @classmethod
-    def run(cls, args=None, standalone=False):
-        """invoke the doit application as a standalone command line tool."""
-        # parse the argument we know and pass the rest to the doit application.
-        ns, args = cls.parse_args(args)
-        main = cls(**ns.__dict__)
-
-        main.main(args, standalone=standalone)
-
-    @classmethod
-    def get_parser(cls):
-        """the argument parser for the Main cli"""
-        from argparse import ArgumentParser
-
-        parser = ArgumentParser("builder")
-        parser.add_argument("-i", "--ids", default=["jupyterlab"], help="repository ids", nargs="*")
-        parser.add_argument(
-            "-d", "--dir", default=Path("jupyter-ally-build"), help="the build directory", type=Path
-        )
-        return parser
-
-    @classmethod
-    def parse_args(cls, args=None):
-        parser = cls.get_parser()
-        return parser.parse_known_args(args=prep_args(args))
-
-
-class Tasks(Base):
-    def doit(self):
-        """build the doit application by pulling tasks from the"""
-        from doit.cmd_base import ModuleTaskLoader
-        from doit.doit_cmd import DoitMain
-
-        tasks = dict((x, getattr(self, x)) for x in dir(self) if x.startswith("task_"))
-        tasks.update({"DOIT_CONFIG": DOIT_CONFIG})
-        return DoitMain(ModuleTaskLoader(tasks))
-
-    def commands(self):
-        """explore what doit knows"""
-        return self.doit().get_commands()
-
-    def do(self, x, **kwargs):
-        """create a cmd action that can be run from the api"""
-        return do(f"{self.conda} {x}", **kwargs)
-
-
-def prep_args(x):
-    if isinstance(x, str):
-        from shlex import split
-
-        return split(x)
-    return x
+##############################################################################################
+## Tasks
+##############################################################################################
+
+
+build_dir = ROOT / "build"
+conda_env_dir = build_dir / ".env"
+prefix = conda_env_dir.resolve()
+conda = f"conda run --no-capture-output --prefix {prefix}"
+
+def task_create_env():
+    """Create a test environment into which dependencies needed by the test code will be installed"""
+    yield dict(
+        name="conda",
+        actions=[
+            do(
+                f'conda create -yc conda-forge --prefix {prefix} python=3.9 "nodejs>=14.18,<15" yarn git'
+            ),
+            do(f"{conda} python -m pip install pip --upgrade"),
+        ],
+        uptodate=[prefix.exists()],
+        clean=[(remove_directory, [prefix])],
+    )
+
+name = "jupyterlab"
```

Comment:
Why are you defining this variable as global?

Reply:
Just a stop gap. My first goal was to get the code working as quickly as possible.